lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/driver/AzureBatchResourceManager.java

```diff
@@ -45,54 +45,89 @@
   private static final Logger LOG = Logger.getLogger(AzureBatchResourceManager.class.getName());
 
   private final Map<String, ResourceRequestEvent> containerRequests;
-
+  private final Set<String> activeContainerIds;
   private final ConfigurationSerializer configurationSerializer;
   private final AzureBatchEvaluatorShimManager evaluatorShimManager;
+  private final AzureBatchTaskStatusAlarmHandler azureBatchTaskStatusAlarmHandler;
 
   private final double jvmHeapFactor;
   private final CommandBuilder launchCommandBuilder;
+  private final AtomicInteger containerCount;
 
   @Inject
   AzureBatchResourceManager(
       final ConfigurationSerializer configurationSerializer,
       final CommandBuilder launchCommandBuilder,
       final AzureBatchEvaluatorShimManager evaluatorShimManager,
+      final AzureBatchTaskStatusAlarmHandler azureBatchTaskStatusAlarmHandler,
       @Parameter(JVMHeapSlack.class) final double jvmHeapSlack) {
     this.configurationSerializer = configurationSerializer;
     this.evaluatorShimManager = evaluatorShimManager;
     this.jvmHeapFactor = 1.0 - jvmHeapSlack;
     this.launchCommandBuilder = launchCommandBuilder;
     this.containerRequests = new ConcurrentHashMap<>();
+    this.activeContainerIds = Collections.synchronizedSet(new HashSet<String>());
+    this.containerCount = new AtomicInteger(0);
+    this.azureBatchTaskStatusAlarmHandler = azureBatchTaskStatusAlarmHandler;
   }
 
   public void onResourceRequested(final ResourceRequestEvent resourceRequestEvent) {
     LOG.log(Level.FINEST, "Got ResourceRequestEvent in AzureBatchResourceManager,");
     for (int r = 0; r < resourceRequestEvent.getResourceCount(); r++) {
       final String containerId = generateContainerId();
       this.containerRequests.put(containerId, resourceRequestEvent);
+      this.containerCount.incrementAndGet();
+      this.azureBatchTaskStatusAlarmHandler.enableAlarm();
```

Comment:
should this enableAlarm() only be called once, if resourceRequestEvent.getResourceCount() > 0? Since we disable the alarm only when it is <=0

Reply:
We are simply saying that everytime a resource is requested, the alarm must be enabled .. so that it can send back status. Also, since it is just a boolean, enabling it multiple times is the same thing as enabling it once. Won't fix.